### PR TITLE
feat: api of DingTalk notification bot

### PR DESCRIPTION
**Pull request by aFlyBird0**
Signed-off-by: Bird <aflybird0@gmail.com>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
Add DingTalk bots into Jenkins to support DingTalk Notifications.

## Related Issues

Core Part of step 2 in #1086

### 1. add one bot

**Code**:
image

**Result**:
image

### 2. add two new bots and override the first bot

**Code**:
image

**Result**:
image

### CODE REVIEW

**Comment by aFlyBird0** [comment]
I removed the `json` sample and `example_test.go`, which you can find in the commit history.

**Comment by aFlyBird0** [comment]
> you don't install this plugin in code, maybe you can refer to `pkg/util/jenkins/plugins.go` for more info

You wrote the main entrance to this last feature, so I think it's more appropriate for you to complete it. I've marked the id of the plugin in the comments, but of course I can write it, it's just a function call.

File: pkg/util/jenkins/dingtalk.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	// send request
+	ctx := context.Background()
+	res, err := j.Requester.Post(ctx, "/manage/dingtalk/configure", nil, nil, params)
+
+	if err != nil {
+		return fmt.Errorf("apply dingtalk bot failed: %v", err)
+	}
+	if res.StatusCode != 200 {
```
Comment (iyear) [inline pkg/util/jenkins/dingtalk.go:37]: http.StatusOK

File: pkg/util/jenkins/dingtalk.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+		body, _ := io.ReadAll(res.Body)
```
Comment (iyear) [inline pkg/util/jenkins/dingtalk.go:38]: do not ignore error

File: pkg/util/jenkins/dingtalk/type.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package dingtalk
+
+type (
+	InnerBotConfig struct {
+		NoticeOccasions []string       `json:"noticeOccasions"`
+		Verbose         bool           `json:"verbose"`
+		ProxyConfig     ProxyConfig    `json:"proxyConfig"`
+		RobotConfigs    []RobotConfigs `json:"robotConfigs"`
+		CoreApply       string         `json:"core:apply"`
```
Comment (iyear) [inline pkg/util/jenkins/dingtalk/type.go:9]: make sure json tag is split by `:`

File: pkg/util/jenkins/dingtalk.go (hunk 1 above)
Comment (aFlyBird0) [reply pkg/util/jenkins/dingtalk.go:37]: Thank you

**Review by iyear — APPROVED** [review]
LGTM👏

Please test it fully, I can't review dingtalk business logic.

**Review by daniel-hutao — APPROVED** [review]
lgtm

**Pull request merged**